ui/src/app/lib/atlasmap-data-mapper/components/document-definition.component.ts

```diff
@@ -132,9 +139,16 @@ export class DocumentDefinitionComponent implements OnInit {
     }
     return count;
   }
+
+  /**
+   * Handle scrolling in this document definition instance.  Avoid a circular dependence with the
+   * LineMachineComponent by dispatching a custom mappings-line-redraw event.
+   * @param event
+   */
   handleScroll(event: any) {
     this.scrollTop = event.target.scrollTop;
-    this.lineMachine.redrawLinesForMappings();
+    this.redrawMappingLinesEvent = new CustomEvent('redrawMappingLines', {detail: this.lineMachine});
+    document.dispatchEvent(this.redrawMappingLinesEvent);
```

Comment:
oh I just got to know this is a DOM document, built in var in javascript. I'm curious if  it's common way vs. using Angular `EventEmitter`.
https://github.com/syndesisio/syndesis/blob/248a2dbf077f324840b74830f69fe229752ceef1/app/ui/src/app/integration/edit-page/current-flow.service.ts#L352

Reply:
Exactly, although it’s not commonly used outside the Web Workers arena unless one does not rely on any web components framework such as React, Vue or Angular, which is our case.

I rather stick to one single way to dispatch events in order not to mix up different paradigms in a single architecture and therefore I’d stick to Angulars built in output properties, just for consistency sake. :)

Reply:
And this is an example to emit the event, directly invoke `emit()` on a `CurrentFlowService.events`
https://github.com/syndesisio/syndesis/blob/248a2dbf077f324840b74830f69fe229752ceef1/app/ui/src/app/integration/edit-page/step-configure/data-mapper/data-mapper-host.component.ts#L134